src/server/api/routers/assignment/delete.tsx

```diff
@@ -0,0 +1,22 @@
+import { z } from "zod";
+import { protectedProcedure } from "@/server/api/trpc";
+import { TRPCError } from "@trpc/server";
+
+export const deleteAssignment = protectedProcedure
+  .input(z.object({ id: z.string() }))
+  .mutation(async ({ ctx, input }) => {
+    try {
```

Comment:
add check if the id exists or not, or if it is already deleted. Check Tag `delete.tsx`